jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfigFactory.java

```diff
@@ -0,0 +1,230 @@
+  static final String CUSTOM_JMX_SCRAPING_CONFIG = PREFIX + "jmx.custom.jmx.scraping.config";
+  static final String TARGET_SYSTEM = PREFIX + "jmx.target.system";
+  static final String INTERVAL_MILLISECONDS = PREFIX + "jmx.interval.milliseconds";
+  static final String METRICS_EXPORTER_TYPE = PREFIX + "metrics.exporter";
+  static final String EXPORTER_INTERVAL = PREFIX + "metric.export.interval";
+  static final String REGISTRY_SSL = PREFIX + "jmx.remote.registry.ssl";
+
+  static final String OTLP_ENDPOINT = PREFIX + "exporter.otlp.endpoint";
+
+  static final String JMX_USERNAME = PREFIX + "jmx.username";
+  static final String JMX_PASSWORD = PREFIX + "jmx.password";
+  static final String JMX_REMOTE_PROFILE = PREFIX + "jmx.remote.profile";
+  static final String JMX_REALM = PREFIX + "jmx.realm";
+
+  // These properties need to be copied into System Properties if provided via the property
+  // file so that they are available to the JMX Connection builder
+  static final List<String> JAVA_SYSTEM_PROPERTIES =
+      Arrays.asList(
+          "javax.net.ssl.keyStore",
+          "javax.net.ssl.keyStorePassword",
+          "javax.net.ssl.keyStoreType",
+          "javax.net.ssl.trustStore",
+          "javax.net.ssl.trustStorePassword",
+          "javax.net.ssl.trustStoreType");
+
+  static final List<String> AVAILABLE_TARGET_SYSTEMS =
+      Arrays.asList(
+          "activemq",
+          "cassandra",
+          "hbase",
+          "hadoop",
+          "jetty",
+          "jvm",
+          "kafka",
+          "kafka-consumer",
+          "kafka-producer",
+          "solr",
+          "tomcat",
+          "wildfly");
+
+  private Properties properties = new Properties();
+
+  /**
+   * Create {@link JmxScraperConfig} object basing on command line options
+   *
+   * @param args application commandline arguments
+   */
+  public JmxScraperConfig createConfigFromArgs(List<String> args) {
+    if (!args.isEmpty() && (args.size() != 2 || !args.get(0).equalsIgnoreCase("-config"))) {
+      System.out.println(
+          "Usage: java io.opentelemetry.contrib.jmxscraper.JmxScraper "
+              + "-config <path_to_config.properties or - for stdin>");
+      System.exit(1);
+    }
+
+    Properties loadedProperties = new Properties();
+    if (args.size() == 2) {
+      String path = args.get(1);
+      if (path.trim().equals("-")) {
+        loadPropertiesFromStdin(loadedProperties);
+      } else {
+        loadPropertiesFromPath(loadedProperties, path);
+      }
+    }
+
+    JmxScraperConfig config = createConfig(loadedProperties);
+    validateConfig(config);
+    populateJmxSystemProperties();
+
+    return config;
+  }
+
+  private static void loadPropertiesFromStdin(Properties props) {
+    try (InputStream is = new DataInputStream(System.in)) {
+      props.load(is);
+    } catch (IOException e) {
+      System.out.println("Failed to read config properties from stdin: " + e.getMessage());
+      System.exit(1);
```

Comment:
I think it's better to keep the calls to `System.exit` in `main` and propagate `IOException` in the method signature (or wrap it into a `ConfigurationException`), otherwise that makes this code impossible to unit test.